tests/aws/services/events/test_events_integrations.py

```diff
@@ -0,0 +1,483 @@
+            "BucketARN": arns.s3_bucket_arn(s3_bucket),
+            "Prefix": s3_prefix,
+        },
+    )
+    stream_arn = stream["DeliveryStreamARN"]
+
+    aws_client.events.create_event_bus(Name=bus_name)
+    aws_client.events.put_rule(
+        Name=rule_name,
+        EventBusName=bus_name,
+        EventPattern=json.dumps(TEST_EVENT_PATTERN),
+    )
+    rs = aws_client.events.put_targets(
+        Rule=rule_name,
+        EventBusName=bus_name,
+        Targets=[{"Id": target_id, "Arn": stream_arn}],
+    )
+
+    assert "FailedEntryCount" in rs
+    assert "FailedEntries" in rs
+    assert rs["FailedEntryCount"] == 0
+    assert rs["FailedEntries"] == []
+
+    aws_client.events.put_events(
+        Entries=[
+            {
+                "EventBusName": bus_name,
+                "Source": TEST_EVENT_PATTERN["source"][0],
+                "DetailType": TEST_EVENT_PATTERN["detail-type"][0],
+                "Detail": json.dumps(EVENT_DETAIL),
+            }
+        ]
+    )
+
+    # run tests
+    bucket_contents = aws_client.s3.list_objects(Bucket=s3_bucket)["Contents"]
+    assert len(bucket_contents) == 1
+    key = bucket_contents[0]["Key"]
+    s3_object = aws_client.s3.get_object(Bucket=s3_bucket, Key=key)
+    actual_event = json.loads(s3_object["Body"].read().decode())
+    assert_valid_event(actual_event)
+    assert actual_event["detail"] == EVENT_DETAIL
+
+    # clean up
+    aws_client.firehose.delete_delivery_stream(DeliveryStreamName=stream_name)
+    # empty and delete bucket
+    aws_client.s3.delete_object(Bucket=s3_bucket, Key=key)
+    aws_client.s3.delete_bucket(Bucket=s3_bucket)
+    clean_up(bus_name=bus_name, rule_name=rule_name, target_ids=target_id)
+
+
+@markers.aws.unknown
+def test_put_events_with_target_kinesis(aws_client):
+    rule_name = "rule-{}".format(short_uid())
+    target_id = "target-{}".format(short_uid())
+    bus_name = "bus-{}".format(short_uid())
+    stream_name = "stream-{}".format(short_uid())
+    stream_arn = arns.kinesis_stream_arn(stream_name, TEST_AWS_ACCOUNT_ID, TEST_AWS_REGION_NAME)
+
+    aws_client.kinesis.create_stream(StreamName=stream_name, ShardCount=1)
+
+    aws_client.events.create_event_bus(Name=bus_name)
+
+    aws_client.events.put_rule(
+        Name=rule_name,
+        EventBusName=bus_name,
+        EventPattern=json.dumps(TEST_EVENT_PATTERN),
+    )
+
+    put_response = aws_client.events.put_targets(
+        Rule=rule_name,
+        EventBusName=bus_name,
+        Targets=[
+            {
+                "Id": target_id,
+                "Arn": stream_arn,
+                "KinesisParameters": {"PartitionKeyPath": "$.detail-type"},
+            }
+        ],
+    )
+
+    assert "FailedEntryCount" in put_response
+    assert "FailedEntries" in put_response
+    assert put_response["FailedEntryCount"] == 0
+    assert put_response["FailedEntries"] == []
+
+    def check_stream_status():
```

Comment:
Just an idea for later: Such waiters seem quite common in our test suite. We could consider adding a re-usable helper or even waiter (Dominik [demoed](https://www.notion.so/localstack/df52608d6a4542a1a0da7a94caf63e4e?v=8f97db112f074550b3600cf0e466856d&pvs=4) how the boto waiters can be extended).